server/war/src/main/webapp/WEB-INF/web.xml

```diff
@@ -33,6 +33,8 @@
     <web-resource-collection>
       <web-resource-name>hawkular-apm-server</web-resource-name>
       <url-pattern>/*</url-pattern>
+      <!-- allow OPTIONS for CORS -->
+      <http-method-omission>OPTIONS</http-method-omission>

```

Comment:
This also means that any other OPTIONS endpoint we might have is unsecured. I don't think we have any other OPTIONS endpoint, though, so, this wouldn't be a big problem. 

In general, I think a CORS filter doesn't belong to APM. I'd rather implement this on commons (if there's no similar filter there already), or see if there's a library somewhere that could be used. I know that Keycloak uses it, and it's integrated with the security stack, but not sure it can be reused.